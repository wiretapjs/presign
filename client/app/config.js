export default {
  registrationLink: undefined,
  environment: undefined,
  libertySubDomainMapping: undefined,
  visitor: undefined,
}
