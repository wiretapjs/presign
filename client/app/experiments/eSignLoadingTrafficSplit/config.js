import { isTestEnvironment } from 'app/experiments/configHelpers'
import Variant from '../Variant'

export default function () {
  return {
    salt: 'eSignLoadTrafficSplit',
    percentBaseVersion: isTestEnvironment() ? 0 : 50,
    base: new Variant('eSignLoadingSplit-redirect-to-eService'),
    treatmentA: new Variant('eSingLoadingSplit-redirect-to-our-Loader'),
  }
}
