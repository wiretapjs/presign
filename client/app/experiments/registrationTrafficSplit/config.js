import { isTestEnvironment } from 'app/experiments/configHelpers'
import Variant from '../Variant'

export default function () {
  return {
    salt: 'registrationTrafficSplit',
    percentBaseVersion: isTestEnvironment() ? 0 : 99,
    base: new Variant('regSplit-base'),
    treatmentA: new Variant('regSplit-a'),
  }
}
