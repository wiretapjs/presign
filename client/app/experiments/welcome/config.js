import Variant from '../Variant'

export default {
  salt: 'welcome',
  percentBaseVersion: 50,
  base: new Variant('base'),
  treatmentA: new Variant('a'),
}
