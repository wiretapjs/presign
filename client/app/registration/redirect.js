import getESignDestination from 'app/esign/loading/destination'
import { getEsignLoginDestination, getESignRegDestination } from './destination'

class Redirect {
  eserviceEsignLogin (handstampSuffix) {
    window.location.href = getEsignLoginDestination(handstampSuffix)
  }

  eserviceEsign (handstampSuffix) {
    window.location.href = getESignDestination(handstampSuffix)
  }

  eserviceEsignRegister (handstampSuffix) {
    window.location.href = getESignRegDestination(handstampSuffix)
  }
}

export default new Redirect()
