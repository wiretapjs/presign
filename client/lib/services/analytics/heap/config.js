export default {
  welcome: {
    a: 'IJBAP | Welcome | View PB-A',
    base: 'IJBAP | Welcome | View PB-BASE',
  },
  initializeRegistration: 'IJBAP | Registration | View',
  getStartedClick: {
    a: 'IJBAP | Welcome | Click | Get Started PB-A',
    base: 'IJBAP | Welcome | Click | Get Started PB-Base',
  },
  submitRegistrationClick: 'IJBAP | Registration | Click Submit',
  completeRegistration: 'IJBAP | Registration | Complete',
  completeAutomaticLogin: 'IJBAP | Registration | Automatic Login',
  welcomeScreen: 'IJBAP | Completion',
  eSignDocumentScreen: {
    cantSeeDocsLink: "IJBAP | Esign | Click | Can't See Your Documents",
    needHelp: 'IJBAP | Esign | Click | Need Help',
  },
}
