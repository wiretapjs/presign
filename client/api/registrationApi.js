import fetchApi from 'lib/services/fetch/fetchApi'
import config from 'app/config'
import QueryString from 'query-string'

class RegistrationApi {
  constructor () {
    this.requestUserInfo = null
  }

  async submitRegistration (requestObject) {
    const defaultRequestObj = {
      registrationKey: config.visitor.regToken,
      autoRegReporting: this.getRegistrationOriginCode(),
    }
    Object.assign(defaultRequestObj, requestObject)
    const response = await fetchApi.request('/api/registration/v1/registerUserUsingRegToken', {body: defaultRequestObj})

    if (response) return response

    throw new Error('Registration response was not received')
  }

  getRegistrationOriginCode () {
    return this._isRepAssistedReg ? 'RA' : 'EA'
  }

  get _isRepAssistedReg () {
    const parsedParams = QueryString.parse(window.location.search)
    return parsedParams.src && parsedParams.src.includes('eservice_automaticregistration_createaccount')
  }

  async getUserInfo (analyticsIdentifyUser) {
    if (this.requestUserInfo) return this.requestUserInfo
    const response = await fetchApi.request(`/api/registration/v1/registeredUser/${config.visitor.regToken}`)
    this.requestUserInfo = response
    const { emailAddress, policyNumber } = this.requestUserInfo.registrationInfo
    if (typeof analyticsIdentifyUser === 'function' && emailAddress && policyNumber) {
      analyticsIdentifyUser({emailAddress, policyNumber})
    }
    return response
  }
}
export default new RegistrationApi()
