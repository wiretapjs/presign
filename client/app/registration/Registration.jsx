import React, { Component } from 'react'
import PropTypes from 'prop-types'
import logoImg from 'lib/resources/images/LMI-logo.png'
import { Logo } from 'lib/components'
import autoBind from 'react-autobind'
import registrationApi from 'api/registrationApi'
import CoreLayout from 'app/CoreLayout'
import redirect from 'lib/util/redirect'
import RegistrationFormComponent from './RegistrationFormComponent'
import RegistrationLoading from './RegistrationLoading'
import RegistrationTokenError from './RegistrationTokenError'
import { Card, RegContainer, SubheadlineContainer, HeadlineContainer } from './registration.style'

const regContainer = {
  fontFamily: 'Roboto, sans-serif',
}

const logoContainer = {
  paddingTop: 10,
  paddingBottom: 10,
  textAlign: 'center',
}
const termsText = {
  fontSize: 10,
  marginTop: 15,
  textAlign: 'center',
  color: '#969696',
}
const cardContentContainer = {
  textAlign: 'left',
}

class Registration extends Component {
  constructor (props) {
    super(props)
    this.state = {
      showLoading: true,
      emailAddress: null,
      displaySsn: true,
      showEmailError: false,
    }
    autoBind(this)
  }

  componentDidMount () {
    registrationApi.getUserInfo().then((response) => {
      if (response.isValidToken) {
        this.validTokenHandler(response)
      } else {
        this.inValidTokenHandler()
      }
    }).catch(() => {
      this.inValidTokenHandler()
    })
  }

  validTokenHandler (response) {
    const { userAlreadyRegistered, registrationKeyActive } = response.registrationInfo
    if (userAlreadyRegistered) {
      redirect.eserviceEsignLogin()
    } else if (!registrationKeyActive) {
      redirect.eserviceEsignRegister()
    } else {
      this.setState({
        showLoading: false,
        emailAddress: response.registrationInfo.emailAddress,
        displaySsn: response.registrationInfo.namedInsuredMissingSSN,
      })
    }
  }

  inValidTokenHandler () {
    this.setState({
      showLoading: false,
      showEmailError: true,
    })
    redirect.eserviceEsignRegister()
  }

  getDisplay () {
    if (this.state.showLoading) {
      return <RegistrationLoading />
    } else if (this.state.showEmailError) {
      return <RegistrationTokenError />
    } else {
      return <RegistrationFormComponent {...this.state} {...this.props} />
    }
  }

  render () {
    return (
      <CoreLayout>
        <div>
          <RegContainer style={regContainer}>
            <div>
              <div style={logoContainer}>
                <Logo src={logoImg} />
              </div>
              <Card>
                <div style={cardContentContainer}>
                  <SubheadlineContainer>
                  Get set up and on your way in a few simple steps!
                  </SubheadlineContainer>
                  <HeadlineContainer>
                  Create your account online
                  </HeadlineContainer>
                  {this.getDisplay()}
                </div>
              </Card>
              <div style={termsText}>
              We only use this information to securely verify your identity.
              Liberty Mutual respects your privacy and will not share your information.
              </div>
            </div>
          </RegContainer>
        </div>
      </CoreLayout>
    )
  }
}

Registration.propTypes = {
  regToken: PropTypes.string,
  email: PropTypes.string,
  displaySsn: PropTypes.bool,
}

export default Registration
