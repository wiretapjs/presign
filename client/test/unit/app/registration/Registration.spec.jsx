import React from 'react'
import { shallow } from 'enzyme'
import Registration from 'app/registration/Registration'
import registrationApi from 'api/registrationApi'
import config from 'app/config'
import { flushPromises } from 'test/helpers/promises'
import redirect from 'lib/util/redirect'

config.registrationLink = 'test'
redirect.eserviceEsignLogin = jest.fn()
redirect.eserviceEsignRegister = jest.fn()

describe('Registration', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should render correctly with loading animation', () => {
    const registrationWrapper = shallow(<Registration />)
    registrationWrapper.setState({ showLoading: true })
    expect(registrationWrapper).toMatchSnapshot()
  })

  it('should render correctly without loading animation', () => {
    const registrationWrapper = shallow(<Registration />)
    registrationWrapper.setState({ showLoading: false })
    expect(registrationWrapper).toMatchSnapshot()
  })

  it('should set user email when mounting', async () => {
    registrationApi.getUserInfo = jest.fn(() => Promise.resolve({isValidToken: true, registrationInfo: { emailAddress: 'a@b.c', namedInsuredMissingSSN: true, registrationKeyActive: true }}))
    const registrationWrapper = shallow(<Registration />)
    const registrationInstance = registrationWrapper.instance()
    await registrationInstance.componentDidMount()
    expect(registrationInstance.state.emailAddress).toEqual('a@b.c')
  })

  it('should show error when mounted if token lookup fails', async () => {
    registrationApi.getUserInfo = jest.fn(() => Promise.reject(new Error('test error handling')))
    const registrationWrapper = shallow(<Registration />)
    const registrationInstance = registrationWrapper.instance()
    await registrationInstance.componentDidMount()
    expect(registrationWrapper).toMatchSnapshot()
  })

  describe('After mounting', () => {
    const mountRegistrationComponent = async () => {
      const registrationWrapper = shallow(<Registration />)
      const registrationInstance = registrationWrapper.instance()
      registrationInstance.componentDidMount()
      await flushPromises()
      return registrationInstance
    }

    it('should display the SSN field after mounting and the user is missing SSN', async () => {
      registrationApi.getUserInfo = jest.fn(() => Promise.resolve({isValidToken: true, registrationInfo: { emailAddress: 'a@b.c', namedInsuredMissingSSN: true, registrationKeyActive: true }}))
      const registrationInstance = await mountRegistrationComponent()
      expect(registrationInstance.state.displaySsn).toEqual(true)
    })

    it('should not display the SSN field after mounting and the user has SSN', async () => {
      registrationApi.getUserInfo = jest.fn(() => Promise.resolve({isValidToken: true, registrationInfo: { emailAddress: 'a@b.c', namedInsuredMissingSSN: false, registrationKeyActive: true }}))
      const registrationInstance = await mountRegistrationComponent()
      expect(registrationInstance.state.displaySsn).toEqual(false)
    })

    it('should redirect to EService Esign Login if user already registered', async () => {
      registrationApi.getUserInfo = jest.fn(() => {
        return Promise.resolve({isValidToken: true,
          registrationInfo: { emailAddress: 'a@b.c',
            userAlreadyRegistered: true,
            namedInsuredMissingSSN: false,
            registrationKeyActive: true,
          }})
      })
      await mountRegistrationComponent()
      expect(redirect.eserviceEsignLogin).toHaveBeenCalledTimes(1)
    })

    it('should redirect to EService Registration if user provide !registrationKeyActive', async () => {
      registrationApi.getUserInfo = jest.fn(() => {
        return Promise.resolve({isValidToken: true,
          registrationInfo: { emailAddress: 'a@b.c',
            userAlreadyRegistered: false,
            namedInsuredMissingSSN: false,
            registrationKeyActive: false,
          }})
      })
      await mountRegistrationComponent()
      expect(redirect.eserviceEsignRegister).toHaveBeenCalledTimes(1)
    })

    it('should redirect to EService EsignReg if registrationKeyActive is not active', async () => {
      registrationApi.getUserInfo = jest.fn(() => {
        return Promise.resolve({ emailAddress: 'a@b.c',
          userAlreadyRegistered: false,
          namedInsuredMissingSSN: false,
          registrationKeyActive: false,
        })
      })
      await mountRegistrationComponent()
      expect(redirect.eserviceEsignRegister).toHaveBeenCalledTimes(1)
    })

    it('should now show loader to false to true if getting User Info fails', async () => {
      registrationApi.getUserInfo = jest.fn(() => Promise.reject(new Error()))
      const registrationInstance = await mountRegistrationComponent()
      expect(registrationInstance.state.showLoading).toEqual(false)
    })
  })
})
