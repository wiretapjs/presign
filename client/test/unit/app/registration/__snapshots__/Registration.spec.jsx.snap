// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration should render correctly with loading animation 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CoreLayout>
    <div>
      <styled.div
        style={
          Object {
            "fontFamily": "Roboto, sans-serif",
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 10,
                "textAlign": "center",
              }
            }
          >
            <Logo
              src="LMI-logo.png"
            />
          </div>
          <styled.div>
            <div
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <styled.div>
                Get set up and on your way in a few simple steps!
              </styled.div>
              <styled.div>
                Create your account online
              </styled.div>
              <RegistrationLoading />
            </div>
          </styled.div>
          <div
            style={
              Object {
                "color": "#969696",
                "fontSize": 10,
                "marginTop": 15,
                "textAlign": "center",
              }
            }
          >
            We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
          </div>
        </div>
      </styled.div>
    </div>
  </CoreLayout>,
  "nodes": Array [
    <CoreLayout>
      <div>
        <styled.div
          style={
            Object {
              "fontFamily": "Roboto, sans-serif",
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                  "textAlign": "center",
                }
              }
            >
              <Logo
                src="LMI-logo.png"
              />
            </div>
            <styled.div>
              <div
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                <styled.div>
                  Get set up and on your way in a few simple steps!
                </styled.div>
                <styled.div>
                  Create your account online
                </styled.div>
                <RegistrationLoading />
              </div>
            </styled.div>
            <div
              style={
                Object {
                  "color": "#969696",
                  "fontSize": 10,
                  "marginTop": 15,
                  "textAlign": "center",
                }
              }
            >
              We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
            </div>
          </div>
        </styled.div>
      </div>
    </CoreLayout>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Registration />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Registration {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "displaySsn": true,
          "emailAddress": null,
          "showEmailError": false,
          "showLoading": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CoreLayout>
          <div>
            <styled.div
              style={
                Object {
                  "fontFamily": "Roboto, sans-serif",
                }
              }
            >
              <div>
                <div
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Logo
                    src="LMI-logo.png"
                  />
                </div>
                <styled.div>
                  <div
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    <styled.div>
                      Get set up and on your way in a few simple steps!
                    </styled.div>
                    <styled.div>
                      Create your account online
                    </styled.div>
                    <RegistrationLoading />
                  </div>
                </styled.div>
                <div
                  style={
                    Object {
                      "color": "#969696",
                      "fontSize": 10,
                      "marginTop": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
                </div>
              </div>
            </styled.div>
          </div>
        </CoreLayout>,
        "_debugID": 2,
        "_renderedOutput": <CoreLayout>
          <div>
            <styled.div
              style={
                Object {
                  "fontFamily": "Roboto, sans-serif",
                }
              }
            >
              <div>
                <div
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Logo
                    src="LMI-logo.png"
                  />
                </div>
                <styled.div>
                  <div
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    <styled.div>
                      Get set up and on your way in a few simple steps!
                    </styled.div>
                    <styled.div>
                      Create your account online
                    </styled.div>
                    <RegistrationLoading />
                  </div>
                </styled.div>
                <div
                  style={
                    Object {
                      "color": "#969696",
                      "fontSize": 10,
                      "marginTop": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
                </div>
              </div>
            </styled.div>
          </div>
        </CoreLayout>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Registration />,
}
`;

exports[`Registration should render correctly without loading animation 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CoreLayout>
    <div>
      <styled.div
        style={
          Object {
            "fontFamily": "Roboto, sans-serif",
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 10,
                "textAlign": "center",
              }
            }
          >
            <Logo
              src="LMI-logo.png"
            />
          </div>
          <styled.div>
            <div
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <styled.div>
                Get set up and on your way in a few simple steps!
              </styled.div>
              <styled.div>
                Create your account online
              </styled.div>
              <RegistrationFormComponent
                displaySsn={true}
                emailAddress={null}
                showEmailError={false}
                showLoading={false}
              />
            </div>
          </styled.div>
          <div
            style={
              Object {
                "color": "#969696",
                "fontSize": 10,
                "marginTop": 15,
                "textAlign": "center",
              }
            }
          >
            We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
          </div>
        </div>
      </styled.div>
    </div>
  </CoreLayout>,
  "nodes": Array [
    <CoreLayout>
      <div>
        <styled.div
          style={
            Object {
              "fontFamily": "Roboto, sans-serif",
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                  "textAlign": "center",
                }
              }
            >
              <Logo
                src="LMI-logo.png"
              />
            </div>
            <styled.div>
              <div
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                <styled.div>
                  Get set up and on your way in a few simple steps!
                </styled.div>
                <styled.div>
                  Create your account online
                </styled.div>
                <RegistrationFormComponent
                  displaySsn={true}
                  emailAddress={null}
                  showEmailError={false}
                  showLoading={false}
                />
              </div>
            </styled.div>
            <div
              style={
                Object {
                  "color": "#969696",
                  "fontSize": 10,
                  "marginTop": 15,
                  "textAlign": "center",
                }
              }
            >
              We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
            </div>
          </div>
        </styled.div>
      </div>
    </CoreLayout>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Registration />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Registration {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "displaySsn": true,
          "emailAddress": null,
          "showEmailError": false,
          "showLoading": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CoreLayout>
          <div>
            <styled.div
              style={
                Object {
                  "fontFamily": "Roboto, sans-serif",
                }
              }
            >
              <div>
                <div
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Logo
                    src="LMI-logo.png"
                  />
                </div>
                <styled.div>
                  <div
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    <styled.div>
                      Get set up and on your way in a few simple steps!
                    </styled.div>
                    <styled.div>
                      Create your account online
                    </styled.div>
                    <RegistrationFormComponent
                      displaySsn={true}
                      emailAddress={null}
                      showEmailError={false}
                      showLoading={false}
                    />
                  </div>
                </styled.div>
                <div
                  style={
                    Object {
                      "color": "#969696",
                      "fontSize": 10,
                      "marginTop": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
                </div>
              </div>
            </styled.div>
          </div>
        </CoreLayout>,
        "_debugID": 4,
        "_renderedOutput": <CoreLayout>
          <div>
            <styled.div
              style={
                Object {
                  "fontFamily": "Roboto, sans-serif",
                }
              }
            >
              <div>
                <div
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Logo
                    src="LMI-logo.png"
                  />
                </div>
                <styled.div>
                  <div
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    <styled.div>
                      Get set up and on your way in a few simple steps!
                    </styled.div>
                    <styled.div>
                      Create your account online
                    </styled.div>
                    <RegistrationFormComponent
                      displaySsn={true}
                      emailAddress={null}
                      showEmailError={false}
                      showLoading={false}
                    />
                  </div>
                </styled.div>
                <div
                  style={
                    Object {
                      "color": "#969696",
                      "fontSize": 10,
                      "marginTop": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
                </div>
              </div>
            </styled.div>
          </div>
        </CoreLayout>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Registration />,
}
`;

exports[`Registration should show error when mounted if token lookup fails 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CoreLayout>
    <div>
      <styled.div
        style={
          Object {
            "fontFamily": "Roboto, sans-serif",
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 10,
                "textAlign": "center",
              }
            }
          >
            <Logo
              src="LMI-logo.png"
            />
          </div>
          <styled.div>
            <div
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <styled.div>
                Get set up and on your way in a few simple steps!
              </styled.div>
              <styled.div>
                Create your account online
              </styled.div>
              <RegistrationLoading />
            </div>
          </styled.div>
          <div
            style={
              Object {
                "color": "#969696",
                "fontSize": 10,
                "marginTop": 15,
                "textAlign": "center",
              }
            }
          >
            We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
          </div>
        </div>
      </styled.div>
    </div>
  </CoreLayout>,
  "nodes": Array [
    <CoreLayout>
      <div>
        <styled.div
          style={
            Object {
              "fontFamily": "Roboto, sans-serif",
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "paddingBottom": 10,
                  "paddingTop": 10,
                  "textAlign": "center",
                }
              }
            >
              <Logo
                src="LMI-logo.png"
              />
            </div>
            <styled.div>
              <div
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                <styled.div>
                  Get set up and on your way in a few simple steps!
                </styled.div>
                <styled.div>
                  Create your account online
                </styled.div>
                <RegistrationLoading />
              </div>
            </styled.div>
            <div
              style={
                Object {
                  "color": "#969696",
                  "fontSize": 10,
                  "marginTop": 15,
                  "textAlign": "center",
                }
              }
            >
              We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
            </div>
          </div>
        </styled.div>
      </div>
    </CoreLayout>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Registration />,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Registration {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "displaySsn": true,
          "emailAddress": null,
          "showEmailError": false,
          "showLoading": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CoreLayout>
          <div>
            <styled.div
              style={
                Object {
                  "fontFamily": "Roboto, sans-serif",
                }
              }
            >
              <div>
                <div
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Logo
                    src="LMI-logo.png"
                  />
                </div>
                <styled.div>
                  <div
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    <styled.div>
                      Get set up and on your way in a few simple steps!
                    </styled.div>
                    <styled.div>
                      Create your account online
                    </styled.div>
                    <RegistrationLoading />
                  </div>
                </styled.div>
                <div
                  style={
                    Object {
                      "color": "#969696",
                      "fontSize": 10,
                      "marginTop": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
                </div>
              </div>
            </styled.div>
          </div>
        </CoreLayout>,
        "_debugID": 8,
        "_renderedOutput": <CoreLayout>
          <div>
            <styled.div
              style={
                Object {
                  "fontFamily": "Roboto, sans-serif",
                }
              }
            >
              <div>
                <div
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Logo
                    src="LMI-logo.png"
                  />
                </div>
                <styled.div>
                  <div
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    <styled.div>
                      Get set up and on your way in a few simple steps!
                    </styled.div>
                    <styled.div>
                      Create your account online
                    </styled.div>
                    <RegistrationLoading />
                  </div>
                </styled.div>
                <div
                  style={
                    Object {
                      "color": "#969696",
                      "fontSize": 10,
                      "marginTop": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  We only use this information to securely verify your identity. Liberty Mutual respects your privacy and will not share your information.
                </div>
              </div>
            </styled.div>
          </div>
        </CoreLayout>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Registration />,
}
`;
